src/fields/Table.php

```diff
@@ -91,6 +92,11 @@ public static function valueType(): string
      */
     public ?array $defaults = [[]];
 
+    /**
+     * @var int|null The highest rowId used by the field's default rows so far
+     */
+    public ?int $maxRowId = null;
```

Comment:
We could probably simplify things and drop `maxRowId` altogether if we just assign random numbers/strings as the row IDs. Maybe UUIDs using `StringHelper::UUID()` / `Craft.uuid()`?